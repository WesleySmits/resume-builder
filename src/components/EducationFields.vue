<template>
    <DynamicList
        :title="getLocalizedString('education')"
        :items="education"
        :defaultItem="defaultEducation"
        :onUpdate="updateEducation"
        :getKey="getKey"
        :direction="'column'"
        :id="`educations`"
    >
        <template #item-fields="{ item, index, updateField, removeItem }">
            <fieldset :id="`education-${index}`">
                <details name="education-curtain" :open="index === 0">
                    <summary>
                        <header>
                            <legend>{{ item.fieldOfStudy || getLocalizedString('newEducation') }}</legend>
                        </header>
                    </summary>
                    <FormField
                        :id="`educationInstitution-${index}`"
                        :label="getLocalizedString('educationInstitution')"
                        :placeholder="getLocalizedString('educationInstitutionPlaceholder')"
                        :helperText="getLocalizedString('educationInstitutionHelperText')"
                        :errorText="getLocalizedString('requiredFieldError')"
                        :modelValue="item.institution"
                        :required="true"
                        @valid="(value) => updateField('institution', value as string)"
                    />

                    <FormField
                        :id="`educationDegree-${index}`"
                        :label="getLocalizedString('educationDegree')"
                        :placeholder="getLocalizedString('educationDegreePlaceholder')"
                        :helperText="getLocalizedString('educationDegreeHelperText')"
                        :errorText="getLocalizedString('requiredFieldError')"
                        :modelValue="item.degree"
                        :required="true"
                        @valid="(value) => updateField('degree', value as string)"
                    />

                    <FormField
                        :id="`educationFieldOfStudy-${index}`"
                        :label="getLocalizedString('educationFieldOfStudy')"
                        :placeholder="getLocalizedString('educationFieldOfStudyPlaceholder')"
                        :helperText="getLocalizedString('educationFieldOfStudyHelperText')"
                        :errorText="getLocalizedString('requiredFieldError')"
                        :modelValue="item.fieldOfStudy"
                        :required="true"
                        @valid="(value) => updateField('fieldOfStudy', value as string)"
                    />

                    <FormField
                        :id="`educationLocation-${index}`"
                        :label="getLocalizedString('educationLocation')"
                        :placeholder="getLocalizedString('educationLocationPlaceholder')"
                        :helperText="getLocalizedString('educationLocationHelperText')"
                        :modelValue="item.location"
                        :required="false"
                        @valid="(value) => updateField('location', value as string)"
                    />

                    <FormField
                        type="date"
                        :id="`educationStartDate-${index}`"
                        :label="getLocalizedString('educationStartDate')"
                        :placeholder="getLocalizedString('educationStartDatePlaceholder')"
                        :helperText="getLocalizedString('educationStartDateHelperText')"
                        :modelValue="item.startDate"
                        :required="false"
                        @valid="(value) => updateField('startDate', value as string)"
                    />

                    <FormField
                        type="date"
                        :id="`educationEndDate-${index}`"
                        :label="getLocalizedString('educationEndDate')"
                        :placeholder="getLocalizedString('educationEndDatePlaceholder')"
                        :helperText="getLocalizedString('educationEndDateHelperText')"
                        :modelValue="item.startDate"
                        :required="false"
                        @valid="(value) => updateField('endDate', value as string)"
                    />

                    <FormField
                        :type="'textarea'"
                        :id="`educationDescription-${index}`"
                        :label="getLocalizedString('educationDescription')"
                        :placeholder="getLocalizedString('educationDescriptionPlaceholder')"
                        :helperText="getLocalizedString('educationDescriptionHelperText')"
                        :modelValue="item.description"
                        :required="false"
                        @valid="(value) => updateField('description', value as string)"
                    />

                    <button @click="removeItem" class="secondary" data-action="remove">
                        {{ getLocalizedString('deleteEducation') }}
                    </button>
                </details>
            </fieldset>
        </template>
    </DynamicList>
</template>

<script setup lang="ts">
import { useResumeStore } from '@/stores/resume';
import { getLocalizedString } from '@/utils/translation';
import FormField from './FormField.vue';
import DynamicList from './DynamicList.vue';

const resumeStore = useResumeStore();
const education = resumeStore.education;

function getKey(item: Education, index: number): string {
    return `${item.institution}-${item.degree}-${index}`;
}

function defaultEducation(): Education {
    return {
        institution: '',
        degree: '',
    };
}

function updateEducation(newEducation: Education[]) {
    resumeStore.setEducation(newEducation);
}
</script>
