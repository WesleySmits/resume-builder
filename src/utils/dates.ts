/**
 * Formats a date into "MMM yyyy" (e.g., "Aug 2024").
 * @param date - The date to format.
 * @param locale - Optional locale string. Defaults to the user's browser locale.
 * @returns A string in the format "MMM yyyy".
 */
export function formatDateToLocale(passedDate: Date | string, locale?: string): string {
    const date = typeof passedDate === 'string' ? parseDateFromString(passedDate) : passedDate;
    const userLocale = locale || navigator.language || 'en-US';

    const options: Intl.DateTimeFormatOptions = {
        year: 'numeric',
        month: 'short',
    };

    return new Intl.DateTimeFormat(userLocale, options).format(date);
}

/**
 * Converts a string in "YYYY-MM-DD" format to a Date object.
 * @param dateString - The date string in "YYYY-MM-DD" format.
 * @returns A Date object or null if the input is invalid.
 */
export function parseDateFromString(dateString: string): Date {
    const [year, month, day] = dateString.split('-').map(Number);

    // Validate the parsed numbers
    if (!year || !month || !day || year < 1 || month < 1 || month > 12 || day < 1 || day > 31) {
        throw new Error(`Invalid date string: ${dateString}`);
    }

    // Create a Date object (note: month is zero-based in JavaScript Date)
    return new Date(year, month - 1, day);
}
