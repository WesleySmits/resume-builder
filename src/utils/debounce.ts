/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
export function debounce(func: Function, wait: number = 750) {
    let timeout: number | undefined
    return function (this: any, ...args: any[]) {
        clearTimeout(timeout)
        timeout = window.setTimeout(() => func.apply(this, args), wait)
    }
}
