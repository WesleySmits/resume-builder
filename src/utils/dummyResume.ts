export function getDummyResume(): ResumeData {
    return {
        general: {
            name: {
                firstName: 'Jon',
                middleName: '',
                lastName: 'Snow',
                displayName: '',
            },
            profilePhoto: 'data:image/jpeg;base64,',
            region: 'The Wall',
            contact: {
                email: 'jon.snow@resume-maker.io',
                phone: '123123123',
            },
            drivingLicense: 'Car',
            functionTitle: 'Watch Commander',
            achievements: ['Defeated the Night King.. \n and his dragon.. \n Sort of', '', 'King in the North'],
            introduction:
                'I am the sword in the darkness. \n I am the watcher on the walls. \n I am the shield that guards the realms',
            colleaguesDescribe: 'Brave',
            colleaguesKnow: 'Loyal',
        },
        skills: {
            languages: ['HTML', 'CSS', 'JavaScript'],
            frameworks: ['Vue.js'],
            platforms: ['Node.js'],
            methodologies: ['Agile'],
            databases: ['MongoDB'],
            tools: ['Git'],
            operatingSystems: ['MacOS'],
        },
        topSkills: [
            {
                name: 'JavaScript',
                yearsOfExperience: 10,
            },
            {
                name: 'TypeScript',
                yearsOfExperience: 6,
            },
        ],
        education: [],
        certifications: [],
        jobs: [],
        personalProjects: [
            {
                title: 'Project 1',
                description: 'This was a very tough 5-minute project.',
                period: {
                    startDate: '2001-01-01',
                    endDate: '2023-01-01',
                },
                skills: {
                    languages: ['JavaScript', 'TypeScript'],
                    frameworks: ['Vue.js', 'React'],
                    platforms: ['Node.js', 'Docker'],
                    methodologies: ['Agile', 'Scrum'],
                    databases: ['MongoDB', 'PostgreSQL'],
                    tools: ['Git', 'Docker'],
                    operatingSystems: ['Windows', 'Linux'],
                },
            },
            {
                title: 'Project 2',
                description: 'This was a very tough 5-minute project.',
                skills: {
                    languages: ['JavaScript', 'TypeScript'],
                    frameworks: ['Vue.js', 'React'],
                    platforms: ['Node.js', 'Docker'],
                    methodologies: ['Agile', 'Scrum'],
                    databases: ['MongoDB', 'PostgreSQL'],
                    tools: ['Git', 'Docker'],
                    operatingSystems: ['Windows', 'Linux'],
                },
            },
        ],
    };
}

export function setupDummyResume(): ResumeData {
    const resumeInitialState = getDummyResume();
    localStorage.setItem('resumeData', JSON.stringify(resumeInitialState));
    return resumeInitialState;
}
