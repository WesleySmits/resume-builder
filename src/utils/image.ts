export function base64ToUint8Array(base64: string): Uint8Array {
    const binaryString = atob(base64.split(',')[1]);
    const len = binaryString.length;
    const bytes = new Uint8Array(len);
    for (let i = 0; i < len; i++) {
        bytes[i] = binaryString.charCodeAt(i);
    }
    return bytes;
}

export function roundImage(base64: string, width: number, height: number): Promise<string> {
    return new Promise((resolve) => {
        const img = new Image();
        img.src = base64;
        img.onload = () => {
            const canvas = document.createElement('canvas');
            canvas.width = width;
            canvas.height = height;
            const ctx = canvas.getContext('2d');

            if (ctx) {
                ctx.clearRect(0, 0, width, height);
                ctx.beginPath();
                ctx.arc(width / 2, height / 2, Math.min(width, height) / 2, 0, Math.PI * 2, true);
                ctx.closePath();
                ctx.clip();

                ctx.drawImage(img, 0, 0, width, height);

                resolve(canvas.toDataURL());
            }
        };
    });
}

export async function convertImageToBase64(file: File): Promise<string> {
    return new Promise((resolve, reject) => {
        try {
            const reader = new FileReader();
            reader.onload = () => resolve(reader.result as string);
            reader.onerror = () => reject(new Error('Failed to convert file to Base64'));
            reader.readAsDataURL(file); // This is where we can inject an error
        } catch (error) {
            reject(new Error(error as string));
        }
    });
}
