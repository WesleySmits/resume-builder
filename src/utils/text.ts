export function capitalize(text: string): string {
    if (!text) {
        return '';
    }

    return text.charAt(0).toUpperCase() + text.slice(1).toLowerCase();
}

export function toCamelCase(text: string): string {
    return text
        .replace(/[^a-zA-Z0-9]/g, ' ')
        .split(' ')
        .map((word, index) => (index === 0 ? word : capitalize(word)))
        .join('');
}
