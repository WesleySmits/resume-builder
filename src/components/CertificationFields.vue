<template>
    <DynamicList
        :title="getLocalizedString('certifications')"
        :items="certifications"
        :defaultItem="defaultCertification"
        :onUpdate="updateCertifications"
        :getKey="getKey"
        :direction="'column'"
    >
        <template #item-fields="{ item, index, updateField, removeItem }">
            <fieldset>
                <details>
                    <summary>
                        <header>
                            <legend>{{ item.title || getLocalizedString('newCertification') }}</legend>
                        </header>
                    </summary>
                    <FormField
                        :id="`certificationTitle-${index}`"
                        :label="getLocalizedString('certificationTitle')"
                        :placeholder="getLocalizedString('certificationTitlePlaceholder')"
                        :helperText="getLocalizedString('certificationTitleHelperText')"
                        :errorText="getLocalizedString('requiredFieldError')"
                        :modelValue="item.title"
                        :required="true"
                        @valid="(value) => updateField('title', value as string)"
                    />

                    <FormField
                        :id="`certificationProvider-${index}`"
                        :label="getLocalizedString('certificationProvider')"
                        :placeholder="getLocalizedString('certificationProviderPlaceholder')"
                        :helperText="getLocalizedString('certificationProviderHelperText')"
                        :errorText="getLocalizedString('requiredFieldError')"
                        :modelValue="item.provider"
                        :required="true"
                        @valid="(value) => updateField('provider', value as string)"
                    />

                    <FormField
                        type="yesno"
                        :id="`certificationCompleted-${index}`"
                        :label="getLocalizedString('certificationCompleted')"
                        :modelValue="item.completed"
                        @valid="(value) => updateField('completed', value as boolean)"
                    />

                    <FormField
                        type="number"
                        :min="1900"
                        :max="new Date().getFullYear()"
                        :id="`certificationYear-${index}`"
                        :label="getLocalizedString('certificationYear')"
                        :placeholder="getLocalizedString('certificationYearPlaceholder')"
                        :helperText="getLocalizedString('certificationYearHelperText')"
                        :modelValue="item.year"
                        :required="false"
                        @valid="(value) => updateField('year', value as number)"
                    />

                    <button @click="removeItem" class="secondary" data-action="remove">
                        {{ getLocalizedString('deleteCertification') }}
                    </button>
                </details>
            </fieldset>
        </template>
    </DynamicList>
</template>

<script setup lang="ts">
import { useResumeStore } from '@/stores/resume';
import { getLocalizedString } from '@/utils/translation';
import FormField from './FormField.vue';
import DynamicList from './DynamicList.vue';

const resumeStore = useResumeStore();
const certifications = resumeStore.certifications;

function getKey(item: Certification, index: number): string {
    return `${item.title}-${item.provider}-${index}`;
}

function defaultCertification(): Certification {
    return {
        title: '',
        provider: '',
        completed: false,
    };
}

function updateCertifications(newCertifications: Certification[]) {
    resumeStore.setCertifications(newCertifications);
}
</script>
