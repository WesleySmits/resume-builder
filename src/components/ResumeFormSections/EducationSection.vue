<template>
    <div class="form">
        <EducationFields />

        <fieldset>
            <legend>{{ getLocalizedString('certifications') }}</legend>
            <p>Fields will be added here</p>
        </fieldset>
    </div>
</template>

<script setup lang="ts">
import { getLocalizedString } from '@/utils/translation';
import EducationFields from '@/components/EducationFields.vue';
</script>
