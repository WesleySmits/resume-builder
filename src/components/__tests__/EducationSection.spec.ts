import { mount, VueWrapper } from '@vue/test-utils';
import { describe, expect, it, vi } from 'vitest';
import EducationSection from '../ResumeFormSections/EducationSection.vue';
import EducationFields from '@/components/EducationFields.vue';
import CertificationFields from '../CertificationFields.vue';
import { createTestingPinia } from '@pinia/testing';
import { getDummyResume } from '@/utils/dummyResume';

describe('EducationSection.vue', () => {
    const resumeInitialState = getDummyResume();

    function getMountedComponent(): VueWrapper {
        return mount(EducationSection, {
            global: {
                plugins: [
                    createTestingPinia({
                        createSpy: vi.fn,
                        initialState: {
                            resume: resumeInitialState,
                        },
                    }),
                ],
            },
        });
    }

    it('renders correctly', () => {
        const wrapper = getMountedComponent();

        expect(wrapper.exists()).toBe(true);

        expect(wrapper.findComponent(EducationFields).exists()).toBe(true);
        expect(wrapper.findComponent(CertificationFields).exists()).toBe(true);
    });

    it('matches the snapshot', () => {
        const wrapper = getMountedComponent();
        expect(wrapper.html()).toMatchSnapshot();
    });
});
