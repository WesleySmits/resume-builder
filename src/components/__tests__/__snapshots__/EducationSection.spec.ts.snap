// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EducationSection.vue > matches the snapshot 1`] = `
"<div class="form">
  <fieldset data-v-7fdf210c="" class="dynamic-list" id="educations">
    <details data-v-7fdf210c="" name="skills-curtain" open="">
      <summary data-v-7fdf210c="">
        <header data-v-7fdf210c="">
          <legend data-v-7fdf210c="">Education</legend>
          <div data-v-7fdf210c="" class="buttons"><button data-v-7fdf210c="" class="primary icon" data-action="add"><svg data-v-7fdf210c="" xmlns="http://www.w3.org/2000/svg" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 6v6m0 0v6m0-6h6m-6 0H6"></path>
              </svg></button>
            <!--v-if-->
          </div>
        </header>
      </summary>
      <ul data-v-7fdf210c="" class="dynamic-list__items">
        <li data-v-7fdf210c="" class="list-item list-item--column">
          <fieldset id="education-0">
            <details name="education-curtain" open="">
              <summary>
                <header>
                  <legend>Healthcare</legend>
                </header>
              </summary>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="educationInstitution-0" id-prop="" class="form-field__input" placeholder="University of Amsterdam" required="" aria-labelledby="educationInstitution-0" aria-describedby="educationInstitution-0-helper" modelvalue="Apple School" value="Apple School"><label data-v-b9645e84="" for="educationInstitution-0">Institution</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the institution you studied at.</small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="educationDegree-0" id-prop="" class="form-field__input" placeholder="Bachelor of Science" required="" aria-labelledby="educationDegree-0" aria-describedby="educationDegree-0-helper" modelvalue="Master of Pears" value="Master of Pears"><label data-v-b9645e84="" for="educationDegree-0">Degree</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the degree you obtained.</small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="educationFieldOfStudy-0" id-prop="" class="form-field__input" placeholder="Computer Science" required="" aria-labelledby="educationFieldOfStudy-0" aria-describedby="educationFieldOfStudy-0-helper" modelvalue="Healthcare" value="Healthcare"><label data-v-b9645e84="" for="educationFieldOfStudy-0">Field of Study</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the field of study.</small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="educationLocation-0" id-prop="" class="form-field__input" placeholder="Amsterdam, Netherlands" aria-labelledby="educationLocation-0" aria-describedby="educationLocation-0-helper" modelvalue="The Stars" value="The Stars"><label data-v-b9645e84="" for="educationLocation-0">Location</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the location of the institution.</small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="educationStartDate-0" id-prop="" class="form-field__input" placeholder="01/2018" aria-labelledby="educationStartDate-0" aria-describedby="educationStartDate-0-helper" modelvalue="1970-01-01" type="date" value="1970-01-01"><label data-v-b9645e84="" for="educationStartDate-0">Start Date</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the start date of your education.</small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="educationEndDate-0" id-prop="" class="form-field__input" placeholder="01/2022" aria-labelledby="educationEndDate-0" aria-describedby="educationEndDate-0-helper" modelvalue="1970-01-01" type="date" value="1970-01-01"><label data-v-b9645e84="" for="educationEndDate-0">End Date</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the end date of your education.</small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field"><textarea data-v-b9645e84="" fieldtype="" id="educationDescription-0" id-prop="" class="form-field__input" placeholder="I studied..." aria-labelledby="educationDescription-0" aria-describedby="educationDescription-0-helper" modelvalue="Extensive study of fruits." type="textarea" value="Extensive study of fruits."><!--v-if--></textarea><label data-v-b9645e84="" for="educationDescription-0">Description</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter a description of your education.</small>
                <!--v-if-->
              </div><button class="secondary" data-action="remove">Delete Education</button>
            </details>
          </fieldset>
        </li>
      </ul>
    </details>
  </fieldset>
  <fieldset data-v-7fdf210c="" class="dynamic-list" id="certifications">
    <details data-v-7fdf210c="" name="certifications-curtain" open="">
      <summary data-v-7fdf210c="">
        <header data-v-7fdf210c="">
          <legend data-v-7fdf210c="">Certifications</legend>
          <div data-v-7fdf210c="" class="buttons"><button data-v-7fdf210c="" class="primary icon" data-action="add"><svg data-v-7fdf210c="" xmlns="http://www.w3.org/2000/svg" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 6v6m0 0v6m0-6h6m-6 0H6"></path>
              </svg></button>
            <!--v-if-->
          </div>
        </header>
      </summary>
      <ul data-v-7fdf210c="" class="dynamic-list__items">
        <li data-v-7fdf210c="" class="list-item list-item--column">
          <fieldset id="certification-0">
            <details>
              <summary>
                <header>
                  <legend>Certification 1</legend>
                </header>
              </summary>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="certificationTitle-0" id-prop="" class="form-field__input" placeholder="AWS Certified Solutions Architect" required="" aria-labelledby="certificationTitle-0" aria-describedby="certificationTitle-0-helper" modelvalue="Certification 1" value="Certification 1"><label data-v-b9645e84="" for="certificationTitle-0">Title</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the title of the certification.</small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="certificationProvider-0" id-prop="" class="form-field__input" placeholder="Amazon Web Services" required="" aria-labelledby="certificationProvider-0" aria-describedby="certificationProvider-0-helper" modelvalue="Provider 1" value="Provider 1"><label data-v-b9645e84="" for="certificationProvider-0">Provider</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the provider of the certification.</small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field">
                <div data-v-40d4be4c="" data-v-b9645e84="" class="yes-no-toggle form-field__input" fieldtype="" id-prop="certificationCompleted-0" required="false" aria-labelledby="certificationCompleted-0" aria-describedby="certificationCompleted-0-helper" type="yesno" value="true"><input data-v-40d4be4c="" type="checkbox" id="certificationCompleted-0"><label data-v-40d4be4c="" for="certificationCompleted-0" class="toggle-label"><span data-v-40d4be4c="" class="toggle-inner checked"></span></label></div><label data-v-b9645e84="" for="certificationCompleted-0">Completed?</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text"></small>
                <!--v-if-->
              </div>
              <div data-v-b9645e84="" class="form-field"><input data-v-b9645e84="" fieldtype="" id="certificationYear-0" id-prop="" class="form-field__input" placeholder="2021" aria-labelledby="certificationYear-0" aria-describedby="certificationYear-0-helper" modelvalue="2021" min="1900" max="2025" type="number" value="2021"><label data-v-b9645e84="" for="certificationYear-0">Year</label><small data-v-b9645e84="" id="\`\${id}-helper\`" class="help-text">Enter the year you completed the certification.</small>
                <!--v-if-->
              </div><button class="secondary" data-action="remove">Delete Certification</button>
            </details>
          </fieldset>
        </li>
      </ul>
    </details>
  </fieldset>
</div>"
`;
